internal/controller/drplacementcontrol_controller.go

```diff
@@ -854,15 +850,76 @@ func (r *DRPlacementControlReconciler) updateAndSetOwner(
 	usrPlacement client.Object,
 	log logr.Logger,
 ) (bool, error) {
-	if err := r.annotateObject(ctx, drpc, usrPlacement, log); err != nil {
+	var (
+		err                       error
+		updated                   bool
+		placementModified         bool
+		placementAnnotateModified bool
+		placementMetadataModified bool
+	)
+
+	// Annotate the placement object
+	if placementAnnotateModified, err = r.annotateObject(drpc, usrPlacement, log); err != nil {
 		return false, err
 	}
 
-	if err := r.updateObjectMetadata(ctx, drpc, usrPlacement, log); err != nil {
+	// Update object metadata
+	if _, placementMetadataModified, err = r.updateObjectMetadata(drpc, usrPlacement); err != nil {
 		return false, err
 	}
 
-	return r.setDRPCOwner(ctx, drpc, usrPlacement, log)
+	// Merge both modifications
+	placementModified = placementAnnotateModified || placementMetadataModified
+
+	// Set DRPC Owner
+	updated, err = r.setDRPCOwner(drpc, usrPlacement, log)
+	if err != nil {
+		return false, err
+	}
+
+	if placementModified {
+		if err = r.updatePlacement(ctx, usrPlacement, log); err != nil {
+			return false, err
+		}
+	}
+
+	if err = r.updateDRPC(ctx, drpc, log); err != nil {
+		return false, err
+	}
```

Comment:
We don't need to check if the drpc needs an update?

Reply:
The DRPC CR updates every time without any condition.

https://github.com/RamenDR/ramen/blob/c975f8c0be43605868319f8427a24cbaaf0a84cc/internal/controller/drplacementcontrol_controller.go#L1042